lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/transport/netty/NettyMessagingTransport.java

```diff
@@ -320,7 +320,7 @@ public void close() {
         }
         break;
       } catch (final Exception e) {
-        if (e.getClass().getSimpleName().compareTo("ConnectException") == 0) {
+        if (e.getClass().getSimpleName().compareTo("AnnotatedConnectException") == 0) {
```

Comment:
Did you actually simulate the "connection refused" event to check the simple name is `AnnotatedConnectException`? 
Can't we do `e instanceof AnnotatedConnectException`?

Reply:
I ran the test and found that the exception's simple name was changed to `connectionException` to `AnnotatedConnecException`. The exception was raised both version of Netty as "connection refused", but exception was changed.

I'll test I can do it as `e instanceof AnnotatedConnectException`. Thanks for your idea!

Reply:
Or you can try the following.
'''
} catch (final AnnotatedConnectException e) {
  ...
} catch (final Exception e) {
  throw e;
}
'''

Reply:
Unfortunately, the `AnnotatedConnectException` is private class of `io.netty.channel.AbstractChannel` and cannot be accessed in public. Could you please tell me other ideas for this?

Reply:
Never mind. I remembered that. Let's go with what you have.